include/cst_tokenstream.h

```diff
@@ -77,7 +77,11 @@ typedef struct cst_tokenstream_struct {
     const cst_string *p_prepunctuationsymbols;
     const cst_string *p_postpunctuationsymbols;
 
+
     cst_string charclass[256];
+    cst_string *charclass2[256];
+    cst_string **charclass3[256];
+    cst_string ***charclass4[256];
 
```

Comment:
I added more comments to document the charclass and the character manipulation a bit better and I reduced the memory usage to make the needed allocations.

In the end we want to allocate a sparse matrix (only few characters belong to "punctuation" or "pre-punctuation"), so the approach I took was to split this sparse matrix in smaller matrices, leveraging on how the UTF-8 characters are encoded. In practical terms, ASCII characters are encoded all together, and the rest are saved in chunks of 64 characters.

This packing makes sense because similar special characters (such as "«" and "»") are also close in their respective UTF-8 encoding.

I don't know if I explained myself well enough...